src/zenml/zen_server/routers/runs_endpoints.py

```diff
@@ -267,3 +284,26 @@ def get_run_status(
         id=run_id, get_method=zen_store().get_run, hydrate=False
     )
     return run.status
+
+
+@router.get(
+    "/{run_id}" + REFRESH,
+    responses={401: error_response, 404: error_response, 422: error_response},
+)
+@handle_exceptions
+def refresh_run_status(
+    run_id: UUID,
+    _: AuthContext = Security(authorize),
+) -> None:
+    """Refreshes the status of a specific pipeline run.
+
+    Args:
+        run_id: ID of the pipeline run to refresh.
+    """
+    # Verify access to the run
+    run = verify_permissions_and_get_entity(
```

Comment:
This should probably also check permissions for the orchestrator?

Reply:
I was considering doing the check here but our RBAC implementation changed my mind. AFAIK, if I don't have access to the stack or the orchestrator within, our RBAC system would make sure that these entity is removed from the `RunResponse`, right? This way, I can check for these cases in the root `RunResponse.refresh_run_status()` method. This way I am also saving 2 calls (for the stack RBAC check and the orchestrator RBAC check). What do you think?

Reply:
Ah I think I get what you mean. All submodels from the response without access will be removed yes, but they will just not have a body and will not be entirely gone.
Additionally, I think the orchestrator is not even part of the run response, and gets loaded via some hydration calls -> RBAC is not used in these cases

Reply:
I see. I implemented it accordingly as an additional check for the `orchestrator`. 

PS: The `stack` is already covered as it is a part of the `RunResponse`.